balancer-js/src/modules/pools/pool-types/concerns/composableStable/exitV3.concern.integration.spec.ts

```diff
@@ -1,172 +1,86 @@
-      );
-      expect(expectedMins).to.deep.eq(minAmountsOut);
+      const bptIn = parseFixed('0.1', 18).toString();
+      await testExactBptIn(pool, signer, bptIn);
     });
   });
 
   context('exitExactTokensOut', async () => {
     it('all tokens with value', async () => {
       const tokensOut = removeItem(pool.tokensList, pool.bptIndex);
       const amountsOut = tokensOut.map((_, i) =>
-        parseFixed((i * 1).toString(), 18).toString()
+        parseFixed(((i + 1) * 0.1).toString(), 18).toString()
       );
-      const slippage = '7';
-      const { to, data, maxBPTIn, expectedBPTIn } =
-        pool.buildExitExactTokensOut(
-          signerAddress,
-          tokensOut,
-          amountsOut,
-          slippage
-        );
-      const { transactionReceipt, balanceDeltas } =
-        await sendTransactionGetBalances(
-          pool.tokensList,
-          signer,
-          signerAddress,
-          to,
-          data
-        );
-      expect(transactionReceipt.status).to.eq(1);
-      const expectedDeltas = insert(amountsOut, pool.bptIndex, expectedBPTIn);
-      expect(expectedDeltas).to.deep.eq(balanceDeltas.map((a) => a.toString()));
-      const expectedMaxBpt = addSlippage(
-        BigNumber.from(expectedBPTIn),
-        BigNumber.from(slippage)
-      ).toString();
-      expect(expectedMaxBpt).to.deep.eq(maxBPTIn);
+      await testExactTokensOut(pool, signer, tokensOut, amountsOut);
     });
     it('single token with value', async () => {
       const tokensOut = removeItem(pool.tokensList, pool.bptIndex);
       const amountsOut = Array(tokensOut.length).fill('0');
-      amountsOut[0] = parseFixed('2', 18).toString();
-      const slippage = '7';
-      const { to, data, maxBPTIn, expectedBPTIn } =
-        pool.buildExitExactTokensOut(
-          signerAddress,
-          tokensOut,
-          amountsOut,
-          slippage
-        );
-      const { transactionReceipt, balanceDeltas } =
-        await sendTransactionGetBalances(
-          pool.tokensList,
-          signer,
-          signerAddress,
-          to,
-          data
-        );
-      expect(transactionReceipt.status).to.eq(1);
-      const expectedDeltas = insert(amountsOut, pool.bptIndex, expectedBPTIn);
-      expect(expectedDeltas).to.deep.eq(balanceDeltas.map((a) => a.toString()));
-      const expectedMaxBpt = addSlippage(
-        BigNumber.from(expectedBPTIn),
-        BigNumber.from(slippage)
-      ).toString();
-      expect(expectedMaxBpt).to.deep.eq(maxBPTIn);
+      amountsOut[0] = parseFixed('0.1', 18).toString();
+      await testExactTokensOut(pool, signer, tokensOut, amountsOut);
+    });
+    it('single token with value, token out after BPT index', async () => {
+      const tokensOut = removeItem(pool.tokensList, pool.bptIndex);
+      const amountsOut = Array(tokensOut.length).fill('0');
+      amountsOut[2] = parseFixed('0.1', 18).toString();
+      await testExactTokensOut(pool, signer, tokensOut, amountsOut);
```

Comment:
Good one! I didn't know about this helper functions for testing

Reply:
Yeah! We're trying to extract and reuse these testing processes so it's more consistent, easier to debug and easier to add new tests. 😃